packages/mgt-chat/src/components/ChatListItem/ChatListItem.tsx

```diff
@@ -0,0 +1,174 @@
+
+  // Copied and modified from the sample ChatItem.tsx
+  // Determines the title in the case of 1:1 and self chats
+  const inferTitle = useCallback(
+    (chatObj: Chat) => {
+      if (myId && chatObj.chatType === 'oneOnOne' && chatObj.members) {
+        const other = chatObj.members.find(m => (m as AadUserConversationMember).userId !== myId);
+        const me = chatObj.members.find(m => (m as AadUserConversationMember).userId === myId);
+        return other
+          ? `${other?.displayName || (other as AadUserConversationMember)?.email || other?.id}`
+          : `${me} (You)`;
+      }
+      return chatObj.topic || chatObj.chatType;
+    },
+    [myId]
+  );
+
+  const extractTimestamp = (timestamp: NullableOption<string> | undefined): string => {
+    if (timestamp === undefined || timestamp === null) return '';
+    const currentDate = new Date();
+    const date = new Date(timestamp);
+
+    const [month, day, year] = [date.getMonth(), date.getDate(), date.getFullYear()];
+    const [currentMonth, currentDay, currentYear] = [
+      currentDate.getMonth(),
+      currentDate.getDate(),
+      currentDate.getFullYear()
+    ];
+
+    if (currentDay === day && currentMonth === month && currentYear === year) {
+      return date.toLocaleTimeString([], { hour: 'numeric', minute: '2-digit' });
+    }
+
+    return date.toLocaleDateString([], { month: 'numeric', day: 'numeric' });
+  };
+
+  const getDefaultProfileImage = () => {
+    // define the JSX for FluentUI Icons + Styling
+    const oneOnOneProfilePicture = <ChatListItemIcon chatType="oneOnOne" />;
+    const GroupProfilePicture = <ChatListItemIcon chatType="group" />;
+
+    switch (true) {
+      case chat.chatType === 'oneOnOne':
+        return oneOnOneProfilePicture;
+      case chat.chatType === 'group':
+        return GroupProfilePicture;
+      default:
+        error(`Error: Unexpected chatType: ${chat.chatType}`);
+        return oneOnOneProfilePicture;
+    }
+  };
+
+  const enrichPreviewMessage = useCallback(
+    (previewMessage: NullableOption<ChatMessageInfo> | undefined) => {
+      if (myId && previewMessage?.from?.user?.id === myId) {
+        return 'You: ' + previewMessage?.body?.content;
+      }
+      if (previewMessage?.from?.user?.displayName) {
+        return previewMessage?.from?.user?.displayName + ': ' + previewMessage?.body?.content;
+      }
+      const membersAddedEventMessageDetail = previewMessage?.eventDetail as MembersAddedEventMessageDetail;
+      if (membersAddedEventMessageDetail) {
+        const initiator = membersAddedEventMessageDetail.initiator;
+        const addedMembers = membersAddedEventMessageDetail.members?.map(m => m.displayName).join(', ');
+        if (myId && initiator?.user?.id === myId) {
+          return `You added ${addedMembers}.`;
+        }
+        return `${initiator?.user?.displayName} added ${addedMembers}.`;
+      }
+
+      return previewMessage?.body?.content;
+    },
+    [myId]
```

Comment:
I don't think there is ever a case that the user's ID could change and the ChatListItem still exist. If the user is logged off, the ChatList should be clearing all ChatListItems anyway.

Reply:
We can do it that way, where the ChatListItems get unmounted on sign out. This way, the content inside can be changed on sign out, and wouldn't necessarily need to be unmounted, they can just update as the inputs change.

Reply:
But the chats that user A has won't be the same as user B.